src/v/archival/ntp_archiver_service.cc

```diff
@@ -818,12 +819,12 @@ ss::future<> ntp_archiver::sync_manifest_until_term_change() {
             vlog(
               _rtclog.error,
               "Failed to download manifest {}",
-              manifest().get_manifest_path());
+              manifest().get_manifest_path(remote_path_provider()));
```

Comment:
Why `remote_path_prvider` is passed into every call? IIUC the `remote_label` is either set for the partition or not. So it'd be possible to pass `remote_path_provider` to the c-tor of the `partition_manifest` in the archival STM (which also "knows" about `remote_label` now).

Reply:
We chatted about this offline and agreed that making the path provider a member of the manifest probably isn't the way to go, given how we use partition_manifest in many places as just a struct that knows about serialization. For now, having a singular remote_path_provider and passing it around is tentatively the least evil.